src/org/rascalmpl/library/IO.rsc

```diff
@@ -327,6 +327,13 @@ Check whether location `file` is actually a file.
 @javaClass{org.rascalmpl.library.Prelude}
 public java bool isFile(loc file);
 
+@synopsis{Check whether a given location is a writable file (and not readonly/non-existing).}
```

Comment:
If this is about file permissions, then should be not also have the other permissions (read, execute) implemented? If we do then this API might look a bit different.

Reply:
The idea is more (and this was a question from 2 different users) how can I prevent an IO exception if I call writeFile on something. The file scheme is writable, but files can still be readonly (for different reasons).

In that direction I can imagine an isReadable, but executable is not something we really expose in our IO api?